File: app/src/main/java/com/wxm/camerajob/data/define/EAction.java
package com.wxm.camerajob.data.define;

import android.support.annotation.Nullable;

/**
 * for action
 * Created by ookoo on 2018/2/21.
 */
public enum EAction {
    LOAD_JOB("load_job"),
    LOAD_CAMERA_SETTING("load_camera_setting"),
    LOAD_PHOTO_DIR("load_photo_dir");

    private String szActionName;

    EAction(String szName)  {
        szActionName = szName;
    }

    /**
     * get properties name
     * @return  name
     */
    public String getName()  {
        return szActionName;
    }

    /**
     * use name to get enum object
     * @param szName   action name
     * @return         enum object
     */
    @Nullable
    public static EAction getEAction(String szName) {
        for(EAction en : EAction.values())  {
            if(en.szActionName.equals(szName))
                return en;
        }

        return null;
    }
}

File: app/src/main/java/com/wxm/camerajob/data/define/EAction.kt
package com.wxm.camerajob.data.define

/**
 * for action
 * Created by WangXM on 2018/2/21.
 */
enum class EAction private constructor(val actName : String) {
    LOAD_JOB("load_job"),
    LOAD_CAMERA_SETTING("load_camera_setting"),
    LOAD_PHOTO_DIR("load_photo_dir");

    companion object {
        /**
         * use paraName to get enum object
         * @param szName   action paraName
         * @return         enum object
         */
        fun getEAction(szName: String): EAction? {
            return EAction.values().find { it.actName == szName }
        }
    }
}
